pkg/context/httpcontext.go

```diff
@@ -206,10 +249,40 @@ func (ctx *httpContext) Span() tracing.Span {
 	return ctx.span
 }
 
+// Fill the empty tag. If all existing Tags are filled, create new one.
 func (ctx *httpContext) AddTag(tag string) {
 	ctx.tags = append(ctx.tags, tag)
 }
 
+// Add new LazyTag.
+func (ctx *httpContext) AddLazyTag(ns string, prefix string, msg string, intMsg int) {
```

Comment:
this tag is not very lazy... lazy usually means the program is only executed when the result is needed. some functional programming languages use this schema a lot...

Reply:
Yes, this LazyTag enables to postpone CPU heavy `stringtools.Cat` and `Itoa` operations away from the execution of the current request. Please note that [LazyTag.String](https://github.com/megaease/easegress/blob/f8ffb32564d589c30d3b71b661fe276ae7cd8d01/pkg/context/httpcontext.go#L180-L186) is executed only [here](https://github.com/megaease/easegress/blob/f8ffb32564d589c30d3b71b661fe276ae7cd8d01/pkg/context/httpcontext.go#L364), lazily. 

You are right that `AddLazyTag` does not avoid allocating the memory needed for the LazyTag struct, but this is where the `lazyTagPool` helps out.